python/ray/rllib/ddpg/ddpg.py

```diff
@@ -0,0 +1,206 @@
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import pickle
+import os
+
+import numpy as np
+import tensorflow as tf
+
+import ray
+from ray.rllib.ddpg.ddpg_evaluator import DDPGEvaluator
+from ray.rllib import optimizers
+from ray.rllib.utils.actors import split_colocated
+from ray.rllib.agent import Agent
+from ray.tune.result import TrainingResult
+
+DEFAULT_CONFIG = dict(
+    # === Model ===
+
+    # Config options to pass to the model constructor
+    model={},
+    # Discount factor for the MDP
+    gamma=0.99,
+    # Arguments to pass to the env creator
+    env_config={},
+
+    # === Exploration ===
+    # Max num timesteps for annealing schedules. Exploration is annealed from
+    # 1.0 to exploration_fraction over this number of timesteps scaled by
+    # exploration_fraction
+    schedule_max_timesteps=100000,
+    # Number of env steps to optimize for before returning
+    timesteps_per_iteration=1000,
+
+    exploration_noise=0.2,
+
+    action_noise=True,
+    # === Replay buffer ===
+    # Size of the replay buffer. Note that if async_updates is set, then
+    # each worker will have a replay buffer of this size.
+    buffer_size=50000,
+
+    # === Optimization ===
+    #
+    tau=0.001,
+    # Learning rate for adam optimizer
+    lr=5e-4,
+    # How many steps of the model to sample before learning starts.
+    learning_starts=1000,
+    # Update the replay buffer with this many samples at once. Note that
+    # this setting applies per-worker if num_workers > 1.
+    sample_batch_size=4,
+    # Size of a batched sampled from replay buffer for training. Note that
+    # if async_updates is set, then each worker returns gradients for a
+    # batch of this size.
+    train_batch_size=32,
+    # Smooth the current average reward over this many previous episodes.
+    smoothing_num_episodes=100,
+
+
+    # === Tensorflow ===
+    # Arguments to pass to tensorflow
+    tf_session_args={
+        "device_count": {"CPU": 2},
+        "log_device_placement": False,
+        "allow_soft_placement": True,
+        "inter_op_parallelism_threads": 1,
+        "intra_op_parallelism_threads": 1,
+    },
+
+    # === Parallelism ===
+    # Number of workers for collecting samples with. This only makes sense
+    # to increase if your environment is particularly slow to sample, or if
+    # you're using the Ape-X optimizer.
+    num_workers=0,
+    # Whether to allocate GPUs for workers (if > 0).
+    num_gpus_per_worker=0,
+    # Optimizer class to use.
+    optimizer_class="LocalSyncOptimizer",
+    # Config to pass to the optimizer.
+    optimizer_config=dict(),
+    # Whether to use a distribution of epsilons across workers for exploration.
+    per_worker_exploration=False,
+    # Whether to compute priorities on workers.
+    worker_side_prioritization=False,
+    # Whether to force evaluator actors to be placed on remote machines.
+    force_evaluators_remote=False)
```

Comment:
Could we drop this config? It used to be used for Ape-X but it has been dropped due to improvements in Ray's actor scheduling.